dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+            hb_room[i + 1]._boundary_condition = bc
+        if self._glazing_ratios is not None:
+            for i, (glz, glz_par) in enumerate(zip(self.glazing_ratios,
+                                                   self.glazing_parameters)):
+                if glz != 0:
+                    if glz_par is None:  # fast scaling operation to make Apertures
+                        face = hb_room[i + 1]
+                        scale_factor = glz ** .5
+                        ap_face = face.geometry.scale(scale_factor, face.geometry.center)
+                        aperture = Aperture(
+                            '{}_Glz1'.format(face.display_name), ap_face)
+                        aperture._parent = face
+                        face._apertures.append(aperture)
+                        # if the Aperture is interior, set adjacent boundary condition
+                        if isinstance(face._boundary_condition, Surface):
+                            names = face._boundary_condition.boundary_condition_objects
+                            adj_ap_name = '{}_Glz1'.format(names[0])
+                            final_names = (adj_ap_name,) + names
+                            aperture.boundary_condition = Surface(final_names, True)
+                    else:  # slower detailed operation to make Apertures
+                        hb_room[i + 1].apertures_by_ratio_rectangle(
+                            glz, glz_par.glazing_height, glz_par.sill_height,
+                            glz_par.horizontal_separation, glz_par.vertical_separation, 0)
+
+        # assign properties for the roof and floor from the parent Story
+        try:
+            hb_room[0].boundary_condition = bcs.adiabatic
+            hb_room[-1].boundary_condition = bcs.adiabatic
+        except AttributeError:
+            pass  # honeybee_energy is not loaded and Adiabatic type doesn't exist
+        if self._parent is not None:
+            if self._parent.is_ground_floor:
+                hb_room[0].boundary_condition = bcs.ground
+            if self._parent.is_top_floor:
+                hb_room[-1].boundary_condition = bcs.outdoors
+
+        return hb_room
+
+    @staticmethod
+    def solve_adjcency(room_polygons, tolerance):
+        """Solve for all adjacencies between a list of input RoomPolygons.
+
+        Args:
+            room_polygons: A list of RoomPolygons for which adjacencies will be solved.
+            tolerance: The minimum difference between the coordinate values of two
+                faces at which they can be considered centered adjacent.
+        """
+        for i, room_1 in enumerate(room_polygons):
+            try:
+                for room_2 in room_polygons[i + 1:]:
+                    for j, seg_1 in enumerate(room_1.floor_segments):
+                        for k, seg_2 in enumerate(room_2.floor_segments):
+                            if not isinstance(room_2._boundary_conditions[k], Surface):
```

Comment:
should you clean all the  boundary conditions before doing this? or at least give the option?

Reply:
Two things:
1. I think you are misinterpreting what this line of code is doing.  This is here to catch the case where the adjacency has already been set by the first Room segment finding its adjacency to the second Room segment (there's no need to re-set the adjacency when the second segment finds the first segment again later in the iteration). So this code is really only here to save a bit of time by not having us do the proximity calculation twice.
2. We can add a method for `clean_boundary_conditions` but it's a little ambiguous what it gets "cleaned" to (ie. outdoors, ground). Also, I'm not sure when I would use it. Can you give me an example of when you need to reset all of the boundary conditions of a Room to be the same? Also is it something that really merits its own method when you can easily do this with a single for loop.

Reply:
Not that I agree with your comments but by reading the rest of comments I think this is good for all we need now. Solving adjacency will most likely be one of the parts that will change several times in the near future.